scripts/pre_commit_linter.py

```diff
@@ -147,6 +147,16 @@
         'excluded_dirs': ()}
 }
 
+BAD_PATTERNS_REGEXP = [
+    {
+        'regexp': r'TODO[^\(]*[^\)][^:]*[^\w]*$',
+        'message': 'Please assign TODO comments to a user ' +
```

Comment:
@apb7, do we want these test to only fail for a specific pattern or for any unexpected TODO pattern?
![image](https://user-images.githubusercontent.com/16653571/61566121-ba5e3d80-aa98-11e9-8db2-0bd201055b0b.png)

Reply:
@DubeySandeep The + sign is used so that each line length is below 80 characters as mandated by pylint

I'm happy to enforce the uppercase starting character in this regexp, and I believe all comments have to end with a period as enforced by one of the linting processes

Reply:
Hi @DubeySandeep, thanks for the comments. Please find the responses inline:

>  do we want these test to only fail for a specific pattern or for any unexpected TODO pattern?

I think the scope of this PR is to ensure that all TODO comments have an associated username in a specific pattern. Let me know if you think otherwise.


> Do we need + here?

Nice catch! No, I don't think we require `+` here. @teddymarchildon, can you please check other multi-line bad pattern messages. They do not use `+`.


> TODO comments end with a period(.) [I think we have such checks for comments]

I think the comment check will cover this as well, right?


> TODO message should start with a capital letter.

I'll file a follow-up issue for this (filed #7182). @teddymarchildon, if you're interested, please let us know on that issue thread. We'll assign it to you. Thanks!

Reply:
Thanks for the reply @apb7, @teddymarchildon can you please update the PR as Apurv suggested so that we can quickly merge this PR?

Reply:
@DubeySandeep I believe I have addressed the comments, and can start working on the next issue once this one is merged. Let me know if there is anything else to fix up or if I missed something from those comments!